### PR TITLE
Query handling

### PR DESCRIPTION
**Pull request by angusbezzina**
This PR addresses some of the loading states to make the app feel less jittery and improve the loading process. 

Not sure if there is much more that I can do at this stage to speed up the queries themselves. In the long term, the goal is to remove GraphQL in favour of ReactQuery so we will only have to manage one layer on top of the application.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "prettier.configPath": "./packages/datum-web/.prettierrc.json",
   "prettier.prettierPath": "./packages/datum-web/node_modules/prettier",
   "editor.defaultFormatter": "esbenp.prettier-vscode",
+  // "editor.formatOnSave": false,
```
Comment (nullaus) [inline .vscode/settings.json:5]: Did you mean to check this in?

**Review by nullaus — COMMENTED** [review]
Looks good! I appreciate the refactoring effort. It feels cleaner now! :) I'm going to test it and circle back with a seal of approval.

File: .vscode/settings.json (hunk 1 above)
Comment (angusbezzina) [reply .vscode/settings.json:5]: Nope, my bad. Let me fix it

**Review by nullaus — APPROVED** [review]
This looks good to me. I did run into the `Users` loading issue we saw earlier though but it's unrelated to these changes, I think?

**Comment by angusbezzina** [comment]
> This looks good to me. I did run into the `Users` loading issue we saw earlier though but it's unrelated to these changes, I think?

Yeah I believe this is unrelated, but I'll investigate further

**Pull request merged**